### ISSUE TITLE
within one.coffee, require('./two') should work like require('./two.coffee!')

### ISSUE THREAD

**Issue by forresto** [forresto/system-coffee#3]
`require('./local.coffee!')` works, but [libraries written in CoffeeScript use](https://github.com/noflo/noflo/blob/ea1800050c5e4d19900be2c22ac5b24962272053/src/lib/NoFlo.coffee#L16) `require('./local')` format. 

@guybedford Is this something that the plugin can suggest?

**Comment by forresto** [comment]
I can imagine a regex on `load.source`, but that might get ugly.

**Comment by guybedford** [comment]
That library would be better off loaded directly through `jspm install npm:...` as the requires are actually npm requires specifically. They may reference Node core libs for example?

I'm not sure you should do something in the plugin? It's worth thinking about.

**Comment by forresto** [comment]
I am trying it with `jspm install npm:noflo` ... I changed jspm_packages/npm/noflo@0.5.11.js to

```
module.exports = require("npm:noflo@0.5.11/src/lib/NoFlo.coffee!");
```

and `System.import('noflo')` works for the main file, but then the internal requires like `require('./Graph')` try to load Graph.js instead of Graph.coffee.

**Comment by guybedford** [comment]
Yes I understand, but the requires are based on the assumption of Node's require pattern, which may be an invalid assumption?

That is, rather than loading existing CoffeeScript, perhaps look at this plugin as for new CoffeeScript?

**Comment by forresto** [comment]
I'd expect `require('./local')` it to work for new CoffeeScript as well. (?)

**Comment by guybedford** [comment]
What about CoffeeScript that contained ES6 syntax (if such a thing were supported ever)?

I think this is CommonJS-specific, that the default extension should match the extension of the current file.

This has helped clarify some decisions I've been needing to make about handling CommonJS in the next iteration of SystemJS. I've added an issue with a suggestion for the next release here - https://github.com/systemjs/systemjs/issues/319.

**Comment by forresto** [comment]
I think that would be a breaking change, require writing `.cs6` or something. Looks like we're on the same page for CommonJS... I'll forget the regex for now ;-)

**Comment by guybedford** [comment]
Yes this is a breaking change, there will be a breaking SystemJS change coming up.

**Comment by forresto** [comment]
CoffeeScript 1.9.0 (released today) does support es6 generators.

``` coffeescript
perfectSquares = ->
  num = 0
  loop
    num += 1
    yield num * num
  return
```
becomes
``` javascript
var perfectSquares;
perfectSquares = function*() {
  var num;
  num = 0;
  while (true) {
    num += 1;
    (yield num * num);
  }
};
```

http://coffeescript.org/#fat-arrow

**Comment by forresto** [comment]
Any changes related to this issue?

> default extension should match the extension of the current file

**Comment by guybedford** [comment]
I'm making the SystemJS updates my primary priority, but I'd suggest not waiting and implementing your own replacements for now if you like.

**Comment by ArmorDarks** [comment]
Hi guys

Is this issue still exists?
